awscli/handlers.py

```diff
@@ -102,6 +103,7 @@ def awscli_initialize(event_handlers):
                             ec2_add_priv_launch_key)
     register_parse_global_args(event_handlers)
     register_pagination(event_handlers)
+    event_handlers.register('operation-args-parsed.ec2.*', set_max_results_default)
```

Comment:
`set_max_results` checks if the function has pagination by looking for `page_size`, which is only present on functions that both have pagination enabled, and also have a limit key. At that point in the stack, I have no access to the pagination config so that's the only way to check. I could potentially manually compile a list of pagination operations, but then that would break whenever EC2 adds more.